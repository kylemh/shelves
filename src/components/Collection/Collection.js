import React from 'react'
import PropTypes from 'prop-types';
import styles from './Collection.module.scss';

export default class Collection extends React.Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
    primaryArtistName: PropTypes.string.isRequired,
    year: PropTypes.number.isRequired,
  };

  render() {
    const { props } = this;

    return (
      <article className={styles.Collection}>
        <h4 className={styles.title}>{props.title}</h4>
        <h5 className={styles.artist}>by {props.primaryArtistName}</h5>
        <h5 className={styles.year}>{props.year}</h5>
      </article>
    )
  }
}
