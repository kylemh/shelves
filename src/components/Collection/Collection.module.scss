@import '../../styles/mixins.scss';
@import '../../styles/variables.scss';

.Collection {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
  padding: 1rem;
  margin: 1rem;
  border-radius: 10px;
  background-color: $white;
  border: 1px solid $primary;
  box-shadow: 0 3px 1px rgba(0, 0, 0, 0.25);
  height: 150px;
  width: 150px;

  @include below-screen-lg {
    height: 125px;
    width: 125px;
  }

  @include below-screen-sm {
    height: 100px;
    width: 100px;
  }

  @include on-hover-and-focus {
    cursor: pointer;
  }
}

.title {
  margin: 0.25rem 0;
  padding: 0;
}

.artist {
  margin: 0;
  padding: 0;
}

.year {
  margin: 0.25rem 0;
  padding: 0;
}
