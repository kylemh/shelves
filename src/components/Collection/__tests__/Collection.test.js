import React from 'react';
import { shallow } from 'enzyme';
import Collection from '../Collection';

describe('Collection', () => {
  const requiredProps = {
    title: 'Cosmogramma',
    primaryArtistName: 'Flying Lotus',
    year: 2010,
  };

  it('should render', () => {
    const wrapper = shallow(<Collection {...requiredProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
