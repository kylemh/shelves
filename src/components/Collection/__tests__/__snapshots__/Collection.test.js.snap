// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection should render 1`] = `
<article
  className="Collection"
>
  <h4
    className="title"
  >
    Cosmogramma
  </h4>
  <h5
    className="artist"
  >
    by 
    Flying Lotus
  </h5>
  <h5
    className="year"
  >
    2010
  </h5>
</article>
`;
